nav2_voxel_grid/include/nav2_voxel_grid/voxel_grid.hpp

```diff
@@ -237,46 +238,60 @@ class VoxelGrid
     int offset_dy = sign(dy) * size_x_;
     int offset_dz = sign(dz);
 
+    // we need to chose how much to scale our dominant dimension, based on the
+    // maximum length of the line
+    double dist = sqrt((x0 - x1) * (x0 - x1) + (y0 - y1) * (y0 - y1) + (z0 - z1) * (z0 - z1));
+    if ((unsigned int)(dist) < min_length) {
+      return;
+    }
+    double scale = std::min(1.0, max_length / dist);
+
+    // Updating x0, y0, z0 to the point at distance min_length from the initial point
+    x0 = x0 + dx / dist * min_length;
```

Comment:
Same here on not re-using variables, it would be good to make new ones clear about what this is